backend/dal/Repositories/DocumentRepository.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using MapsterMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// DocumentRepository class, provides a repository to interact with documents within the datasource.
+    /// </summary>
+    public class DocumentRepository : BaseRepository<PimsDocument>, IDocumentRepository
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a DocumentActivityRepository, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="service"></param>
+        /// <param name="logger"></param>
+        public DocumentRepository(
+            PimsContext dbContext,
+            ClaimsPrincipal user,
+            IPimsRepository service,
+            ILogger<DocumentRepository> logger,
+            IMapper mapper)
+            : base(dbContext, user, service, logger, mapper) { }
+        #endregion
+
+        #region Methods
+
+        public int GetTotalRelationCount(long documentId)
+        {
+            var document = this.Context.PimsDocuments.AsNoTracking()
+                .Include(d => d.PimsActivityInstanceDocuments)
+                .Where(d => d.DocumentId == documentId)
+                .AsNoTracking()
+                .FirstOrDefault();
```

Comment:
nit: I think this query doesn't need two "AsNoTracking()" - isn't it?
Also no need for Where

```cs
var document = this.Context.PimsDocuments.AsNoTracking()
                .Include(d => d.PimsActivityInstanceDocuments)
                .FirstOrDefault(d => d.DocumentId == documentId);

```

Reply:
It was needed since we dont want to make modifications to any of the document information retrieved by the query

Reply:
then how about ?

```cs
this.Context.PimsDocuments
  .Include(d => d.PimsActivityInstanceDocuments)
  .AsNoTracking()
...
```